### ISSUE TITLE
Training Loss Values. 

### ISSUE THREAD

**Issue by mmderakhshani** [showlab/Show-o#50]
Hi, thank you for releasing this GitHub repository. I am trying to reproduce the stage 1 training on ImageNet. Could you please share the W&B log or let me know the initial and final loss values for that stage? I am getting the following loss, and it turns out the model is not converging. Thanks.  

<img width="1353" alt="Screenshot 2024-11-11 at 22 33 42" src="https://github.com/user-attachments/assets/a54819ac-be2c-4311-83da-5f9c473a3e17">

**Comment by Sierkinhane** [comment]
Hi, thanks for your interest. Your loss curve looks a bit weird. When I trained the first stage, the loss would quickly reduce to around 8, and then converge to around 7 (500k steps).
![image](https://github.com/user-attachments/assets/bdcdafb8-2832-4e5d-807a-a75bcab12d1c)

**Comment by Sierkinhane** [comment]
What's the resolution, are you training on 256x256?

**Comment by mmderakhshani** [comment]
I am training with 256x256 resolution. Can you tell at which training steps at stage 1, you are getting meaningful sampled images in class conditional manner? 

One more thing, can you also tell me what is your total batch size and number of training steps in stage 1?

**Comment by Sierkinhane** [comment]
Here are the generated samples at 10k iterations. The total batch size of t2i, mmu, and language modeling is 1152, and I guess the batch size of t2i is around 700.
![image](https://github.com/user-attachments/assets/95f83469-a300-45ff-a63f-8c0f6c92ed5d)

**Comment by mmderakhshani** [comment]
Thanks for the quick reply! Much appreciated.

Can you tell me how would be the generated images at the end of the stage
1. I am assuming in the stage 1, you only are using ImageNet and
RefinedWeb. Is it correct?

PhD student at AIM Lab
Science Park, University of Amsterdam, Amsterdam, Netherlands
Skype Id: m2.derakhshani1993

**Comment by Sierkinhane** [comment]
mmu loss is also used in this stage. Some samples generated without CFG.
![image](https://github.com/user-attachments/assets/d0bee08d-c9eb-4635-a01c-d56f02161e91)

**Comment by mmderakhshani** [comment]
Could you please clarify whether these samples were generated after the training was completed? Additionally, could you let me know how you adjusted the learning rate? Did you use a method similar to YOLO-style initialization, or did you conduct several experiments to determine the best approach? I want to increase the batch size and train a larger model, so I would appreciate some hints about the initialization of the learning rate.

**Comment by Sierkinhane** [comment]
BTW, welcome to star our project :)

**Comment by mmderakhshani** [comment]
Happy to star! :D

**Comment by Sierkinhane** [comment]
Though the loss oscillated around 7 (actually similar to our loss tendency), the image-text alignment was still improving.

**Comment by Sierkinhane** [comment]
It is easier to learn patterns in the synthetic data in the third stage, so the loss will be smaller but still oscillate around some values.

**Comment by dhevarghese** [comment]
Ofcourse :) Just did! 

Thank you for your response.
